File: May-18/08-05-18/python/01-Count-Enemy.py
for _  in range(int(input())):
    data = list(input())
    n = len(data)
    
    flag = False
    ans = 0
    
    for i in range(0, n):
        if(data[i] == 'X'):
            flag = True
        if(data[i] == '*'):
            flag = False
        if(flag and data[i] == 'O'):
            data[i] = '-'
    flag = False
    for i in range(n-1, -1, -1):
        if(data[i] == 'X'):
            flag = True
        if(data[i] == '*'):
            flag = False
        if(flag and data[i] == 'O'):
            data[i] = '-'
    
    ans = data.count('O')
    print(ans)

File: May-18/08-05-18/java/01-Count-Enemy.java

import java.io.*;


class TestClass {
    public static void main(String args[] ) throws Exception {
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(input.readLine());
        while(t-- > 0)
        {
            String str = input.readLine();
            int n = str.length();
            char data[] = new char[n];
            for(int i = 0; i < n; i++)
            {
                data[i] = str.charAt(i);
            }
            
            boolean flag = false;
            int ans = 0;
            
            for(int i = 0; i < n; i++)
            {
                if(data[i] == 'X')
                    flag = true;
                if(data[i] == '*')
                    flag = false;
                if(flag && data[i] == 'O')
                    data[i] = '-';
            }
            flag = false;
            for(int i = n -1; i >= 0; i--)
            {
                if(data[i] == 'X')
                    flag = true;
                if(data[i] == '*')
                    flag = false;
                if(flag && data[i] == 'O')
                    data[i] = '-';
            }
            
            for(int i = 0; i < n; i++)
            {
                if(data[i] == 'O')
                    ans += 1;
            }
            System.out.println(ans);
        }
        
    }
}
